sound/soc/sof/intel/hda.c

```diff
@@ -755,7 +755,7 @@ int hda_dsp_remove(struct snd_sof_dev *sdev)
 				SOF_HDA_PPCTL_GPROCEN, 0);
 
 	free_irq(sdev->ipc_irq, sdev);
-	free_irq(sdev->pci->irq, bus);
+	free_irq(sdev->pci->irq, bus); /* FIXME: is it sdev->hda->irq or sdev->pci->irq ?? */
 	pci_free_irq_vectors(pci);
```

Comment:
ditto.

Reply:
IRQ can be the same physical ID or source for free_irq() as long as the private data pointer are different. Iirc, one of the APL platforms shared the same IRQ ID for both IPC and PCI HDA IRQs.

Reply:
Sorry, I have no idea what this all means. we use both sdev->pci->irq or sdev->hda->irq and it'd be useful to align on a single way of releasing stuff.

Reply:
yes, Pierre, agree on that, I suggest we change this here to sdev->hda->irq.